src/Parsers/ExpressionElementParsers.cpp

```diff
@@ -2316,6 +2317,34 @@ bool ParserOrderByElement::parseImpl(Pos & pos, ASTPtr & node, Expected & expect
     return true;
 }
 
+bool ParserInterpolateElement::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
+{
+    ParserKeyword as("AS");
+    ParserExpression element_p;
+    ParserIdentifier ident_p;
+
+    ASTPtr ident;
+    if (!ident_p.parse(pos, ident, expected))
+        return false;
+
+    ASTPtr expr;
+    if (as.ignore(pos, expected))
+    {
+        if (!element_p.parse(pos, expr, expected))
+            return false;
+    }
+    else
+        expr = ident;
+
+    auto elem = std::make_shared<ASTInterpolateElement>();
+    elem->column = ident->getColumnName();
+    elem->expr = expr;
```

Comment:
You also need to add `expr` to children. Right now, when I do `EXPLAIN AST`, it does not show interpolate child:
```
EXPLAIN AST
SELECT
    n,
    source,
    inter + 1 AS inter_p
FROM
(
    SELECT
        toFloat32(number % 10) AS n,
        'original' AS source,
        number AS inter
    FROM numbers(10)
    WHERE (number % 3) = 1
)
ORDER BY n ASC WITH FILL FROM 0 TO 11.51 STEP 0.5
INTERPOLATE ( inter_p AS inter_p + 1 )

Query id: a0864985-0068-40ae-91d1-64e528442940

┌─explain──────────────────────────────────────────────┐
│ SelectWithUnionQuery (children 1)                    │
...
│    ExpressionList (children 1)                       │
│     InterpolateElement                               │
└──────────────────────────────────────────────────────┘

```

Reply:
done